File: leetcode/AIcoding/4.GPT-4o/c_solutions/1880-check-if-word-equals-summation-of-two-words.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int getValue(char *word) {
    int value = 0;
    for (int i = 0; word[i] != '\0'; i++) {
        value = value * 10 + (word[i] - 'a');
    }
    return value;
}

bool isSumEqual(char *firstWord, char *secondWord, char *targetWord) {
    return getValue(firstWord) + getValue(secondWord) == getValue(targetWord);
}

File: leetcode/AIcoding/3.DeepSeek-v3/rust_solutions/1880-check-if-word-equals-summation-of-two-words.rs
impl Solution {
    pub fn is_sum_equal(first_word: String, second_word: String, target_word: String) -> bool {
        let to_num = |word: String| -> i32 {
            word.chars()
                .map(|c| (c as u8 - b'a') as i32)
                .fold(0, |acc, x| acc * 10 + x)
        };

        let first_num = to_num(first_word);
        let second_num = to_num(second_word);
        let target_num = to_num(target_word);

        first_num + second_num == target_num
    }
}